**Counter-Strike(CS)** is a popular series of tactical first person shooter(FPS) video games that have been enjoyed by the users worldwide.

**Here is the overview of Counter-Strike:**

**GamePlay Overview:**

Counter Strike is primarily a multiplayer game where two teams,the Counter-Terrorists(CTs) and the Terrorists(Ts),compete against each other.

The objective of each round varies based on the game mode, but the primary goals include:

Counter-Terrorists: Prevent the Terrorists from achieving their objectives, such as defusing a bomb or rescuing hostages.

Terrorists: Achieve their objectives, which may include planting a bomb at a designated site or holding hostages. Rounds are relatively short, typically lasting a few minutes, and players have only one life per round. When a player is eliminated, they must wait until the next round to respawn.

Key Features:

Weapons: Players can purchase and use a wide variety of firearms, grenades, and equipment. The choice of weaponry is an essential strategic element in the game.

Economy: Players earn in-game money based on their performance in the previous rounds. Money is used to buy weapons and equipment for the next round.

Maps: Counter-Strike features a range of maps, each with its own layout and objectives. Popular maps include Dust II, Mirage, Inferno, and more.

Teamwork: Successful gameplay in Counter-Strike heavily relies on teamwork, communication, and strategy. Players often coordinate their actions with their teammates to achieve objectives.

Competitive Play: Counter-Strike is well-known for its competitive scene, with professional esports tournaments held worldwide.

Popular Game Modes:

Bomb Defusal (de_): In this mode, Terrorists attempt to plant a bomb at one of the designated bomb sites, while Counter-Terrorists aim to prevent the bomb from being planted or defuse it if it's planted.

Hostage Rescue (cs_): In hostage rescue mode, Counter-Terrorists must rescue hostages held by the Terrorists, while the Terrorists aim to prevent the rescues.

Arms Race: A fast-paced mode where players cycle through a series of weapons, aiming to be the first to get a kill with each weapon.

Deathmatch: A mode where players respawn quickly and aim to get as many kills as possible within a set time limit.

Wingman: A 2v2 competitive mode with smaller maps and shorter rounds.

Counter-Strike has evolved over the years with different versions, including Counter-Strike 1.6, Counter-Strike: Source, and Counter-Strike: Global Offensive (CS:GO), which is the most recent and widely played installment as of my last knowledge update in September 2021.

CS:GO is known for its competitive gameplay, professional esports scene, and ongoing updates that have kept the game relevant and enjoyable for players worldwide. It remains a cornerstone of the first-person shooter genre.

## Problem Statement - Based on previous rounds records we have to build a predictive model which will be predicting which team is going to win a particular round

In [54]:
import numpy as np 
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns 


In [55]:
df = pd.read_csv('/Users/anuragpanchal/Documents/CSgo-Project/Data/csgo_round_snapshots.csv')

In [56]:
df.head()

,time_left,ct_score,t_score,map,bomb_planted,ct_health,t_health,ct_armor,t_armor,ct_money,t_money,ct_helmets,t_helmets,ct_defuse_kits,ct_players_alive,t_players_alive,ct_weapon_ak47,t_weapon_ak47,ct_weapon_aug,t_weapon_aug,ct_weapon_awp,t_weapon_awp,ct_weapon_bizon,t_weapon_bizon,ct_weapon_cz75auto,t_weapon_cz75auto,ct_weapon_elite,t_weapon_elite,ct_weapon_famas,t_weapon_famas,ct_weapon_g3sg1,t_weapon_g3sg1,ct_weapon_galilar,t_weapon_galilar,ct_weapon_glock,t_weapon_glock,ct_weapon_m249,t_weapon_m249,ct_weapon_m4a1s,t_weapon_m4a1s,...,t_weapon_p90,ct_weapon_r8revolver,t_weapon_r8revolver,ct_weapon_sawedoff,t_weapon_sawedoff,ct_weapon_scar20,t_weapon_scar20,ct_weapon_sg553,t_weapon_sg553,ct_weapon_ssg08,t_weapon_ssg08,ct_weapon_ump45,t_weapon_ump45,ct_weapon_xm1014,t_weapon_xm1014,ct_weapon_deagle,t_weapon_deagle,ct_weapon_fiveseven,t_weapon_fiveseven,ct_weapon_usps,t_weapon_usps,ct_weapon_p250,t_weapon_p250,ct_weapon_p2000,t_weapon_p2000,ct_weapon_tec9,t_weapon_tec9,ct_grenade_hegrenade,t_grenade_hegrenade,ct_grenade_flashbang,t_grenade_flashbang,ct_grenade_smokegrenade,t_grenade_smokegrenade,ct_grenade_incendiarygrenade,t_grenade_incendiarygrenade,ct_grenade_molotovgrenade,t_grenade_molotovgrenade,ct_grenade_decoygrenade,t_grenade_decoygrenade,round_winner
0,175.00,0.0,0.0,de_dust2,False,500.0,500.0,0.0,0.0,4000.0,4000.0,0.0,0.0,0.0,5.0,5.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,CT
1,156.03,0.0,0.0,de_dust2,False,500.0,500.0,400.0,300.0,600.0,650.0,0.0,0.0,1.0,5.0,5.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,CT
2,96.03,0.0,0.0,de_dust2,False,391.0,400.0,294.0,200.0,750.0,500.0,0.0,0.0,1.0,4.0,4.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,CT
3,76.03,0.0,0.0,de_dust2,False,391.0,400.0,294.0,200.0,750.0,500.0,0.0,0.0,1.0,4.0,4.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,3.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,CT
4,174.97,1.0,0.0,de_dust2,False,500.0,500.0,192.0,0.0,18350.0,10750.0,0.0,0.0,1.0,5.0,5.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,CT


In [57]:
df.tail()

,time_left,ct_score,t_score,map,bomb_planted,ct_health,t_health,ct_armor,t_armor,ct_money,t_money,ct_helmets,t_helmets,ct_defuse_kits,ct_players_alive,t_players_alive,ct_weapon_ak47,t_weapon_ak47,ct_weapon_aug,t_weapon_aug,ct_weapon_awp,t_weapon_awp,ct_weapon_bizon,t_weapon_bizon,ct_weapon_cz75auto,t_weapon_cz75auto,ct_weapon_elite,t_weapon_elite,ct_weapon_famas,t_weapon_famas,ct_weapon_g3sg1,t_weapon_g3sg1,ct_weapon_galilar,t_weapon_galilar,ct_weapon_glock,t_weapon_glock,ct_weapon_m249,t_weapon_m249,ct_weapon_m4a1s,t_weapon_m4a1s,...,t_weapon_p90,ct_weapon_r8revolver,t_weapon_r8revolver,ct_weapon_sawedoff,t_weapon_sawedoff,ct_weapon_scar20,t_weapon_scar20,ct_weapon_sg553,t_weapon_sg553,ct_weapon_ssg08,t_weapon_ssg08,ct_weapon_ump45,t_weapon_ump45,ct_weapon_xm1014,t_weapon_xm1014,ct_weapon_deagle,t_weapon_deagle,ct_weapon_fiveseven,t_weapon_fiveseven,ct_weapon_usps,t_weapon_usps,ct_weapon_p250,t_weapon_p250,ct_weapon_p2000,t_weapon_p2000,ct_weapon_tec9,t_weapon_tec9,ct_grenade_hegrenade,t_grenade_hegrenade,ct_grenade_flashbang,t_grenade_flashbang,ct_grenade_smokegrenade,t_grenade_smokegrenade,ct_grenade_incendiarygrenade,t_grenade_incendiarygrenade,ct_grenade_molotovgrenade,t_grenade_molotovgrenade,ct_grenade_decoygrenade,t_grenade_decoygrenade,round_winner
122405,15.41,11.0,14.0,de_train,True,200.0,242.0,195.0,359.0,100.0,5950.0,2.0,4.0,1.0,2.0,4.0,0.0,1.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,2.0,1.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,T
122406,174.93,11.0,15.0,de_train,False,500.0,500.0,95.0,175.0,11500.0,23900.0,1.0,2.0,1.0,5.0,5.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,2.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,T
122407,114.93,11.0,15.0,de_train,False,500.0,500.0,495.0,475.0,1200.0,6700.0,3.0,5.0,1.0,5.0,5.0,0.0,2.0,0.0,0.0,0.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,3.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,4.0,4.0,3.0,5.0,1.0,0.0,0.0,5.0,0.0,0.0,T
122408,94.93,11.0,15.0,de_train,False,500.0,500.0,495.0,475.0,1200.0,6700.0,3.0,5.0,1.0,5.0,5.0,0.0,2.0,0.0,0.0,0.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,3.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,1.0,5.0,0.0,3.0,0.0,0.0,0.0,4.0,0.0,0.0,T
122409,74.93,11.0,15.0,de_train,False,375.0,479.0,395.0,466.0,1100.0,7000.0,2.0,5.0,1.0,4.0,5.0,0.0,2.0,0.0,0.0,0.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,1.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,3.0,0.0,2.0,0.0,0.0,0.0,3.0,0.0,0.0,T


In [58]:
df.shape

(122410, 97)

In [59]:
df.columns

Index(['time_left', 'ct_score', 't_score', 'map', 'bomb_planted', 'ct_health',
       't_health', 'ct_armor', 't_armor', 'ct_money', 't_money', 'ct_helmets',
       't_helmets', 'ct_defuse_kits', 'ct_players_alive', 't_players_alive',
       'ct_weapon_ak47', 't_weapon_ak47', 'ct_weapon_aug', 't_weapon_aug',
       'ct_weapon_awp', 't_weapon_awp', 'ct_weapon_bizon', 't_weapon_bizon',
       'ct_weapon_cz75auto', 't_weapon_cz75auto', 'ct_weapon_elite',
       't_weapon_elite', 'ct_weapon_famas', 't_weapon_famas',
       'ct_weapon_g3sg1', 't_weapon_g3sg1', 'ct_weapon_galilar',
       't_weapon_galilar', 'ct_weapon_glock', 't_weapon_glock',
       'ct_weapon_m249', 't_weapon_m249', 'ct_weapon_m4a1s', 't_weapon_m4a1s',
       'ct_weapon_m4a4', 't_weapon_m4a4', 'ct_weapon_mac10', 't_weapon_mac10',
       'ct_weapon_mag7', 't_weapon_mag7', 'ct_weapon_mp5sd', 't_weapon_mp5sd',
       'ct_weapon_mp7', 't_weapon_mp7', 'ct_weapon_mp9', 't_weapon_mp9',
       'ct_weapon_negev', 't_weapon_negev',

In [60]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 122410 entries, 0 to 122409
Data columns (total 97 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   time_left                     122410 non-null  float64
 1   ct_score                      122410 non-null  float64
 2   t_score                       122410 non-null  float64
 3   map                           122410 non-null  object 
 4   bomb_planted                  122410 non-null  bool   
 5   ct_health                     122410 non-null  float64
 6   t_health                      122410 non-null  float64
 7   ct_armor                      122410 non-null  float64
 8   t_armor                       122410 non-null  float64
 9   ct_money                      122410 non-null  float64
 10  t_money                       122410 non-null  float64
 11  ct_helmets                    122410 non-null  float64
 12  t_helmets                     122410 non-nul

Dataset has 122410 rows and 97 columns

bool columns = 1

float64 columns = 94

object columns = 2 ['map','round_winner']

Target Column is **round_winner**

In [61]:
df.isna().sum().sum()

np.int64(0)

Dataset does not have null values 

In [62]:
df.duplicated().sum()

np.int64(4962)

Dataset contains **4962** duplicates rows. We need to drop them.

In [63]:
df.drop_duplicates(inplace=True)

In [64]:
from sklearn.preprocessing import LabelEncoder

We need to convert our object(map,round_winner) and bool(bomb_planted) columns in to number so that they can easily used in lda and model building.

In [65]:
cat_cols = ['map','round_winner','bomb_planted']

for i in cat_cols:
    le= LabelEncoder()
    df[i]= le.fit_transform(df[i])


In [66]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 117448 entries, 0 to 122409
Data columns (total 97 columns):
 #   Column                        Non-Null Count   Dtype  
---  ------                        --------------   -----  
 0   time_left                     117448 non-null  float64
 1   ct_score                      117448 non-null  float64
 2   t_score                       117448 non-null  float64
 3   map                           117448 non-null  int64  
 4   bomb_planted                  117448 non-null  int64  
 5   ct_health                     117448 non-null  float64
 6   t_health                      117448 non-null  float64
 7   ct_armor                      117448 non-null  float64
 8   t_armor                       117448 non-null  float64
 9   ct_money                      117448 non-null  float64
 10  t_money                       117448 non-null  float64
 11  ct_helmets                    117448 non-null  float64
 12  t_helmets                     117448 non-null  fl

In [67]:
df.sample(15)

,time_left,ct_score,t_score,map,bomb_planted,ct_health,t_health,ct_armor,t_armor,ct_money,t_money,ct_helmets,t_helmets,ct_defuse_kits,ct_players_alive,t_players_alive,ct_weapon_ak47,t_weapon_ak47,ct_weapon_aug,t_weapon_aug,ct_weapon_awp,t_weapon_awp,ct_weapon_bizon,t_weapon_bizon,ct_weapon_cz75auto,t_weapon_cz75auto,ct_weapon_elite,t_weapon_elite,ct_weapon_famas,t_weapon_famas,ct_weapon_g3sg1,t_weapon_g3sg1,ct_weapon_galilar,t_weapon_galilar,ct_weapon_glock,t_weapon_glock,ct_weapon_m249,t_weapon_m249,ct_weapon_m4a1s,t_weapon_m4a1s,...,t_weapon_p90,ct_weapon_r8revolver,t_weapon_r8revolver,ct_weapon_sawedoff,t_weapon_sawedoff,ct_weapon_scar20,t_weapon_scar20,ct_weapon_sg553,t_weapon_sg553,ct_weapon_ssg08,t_weapon_ssg08,ct_weapon_ump45,t_weapon_ump45,ct_weapon_xm1014,t_weapon_xm1014,ct_weapon_deagle,t_weapon_deagle,ct_weapon_fiveseven,t_weapon_fiveseven,ct_weapon_usps,t_weapon_usps,ct_weapon_p250,t_weapon_p250,ct_weapon_p2000,t_weapon_p2000,ct_weapon_tec9,t_weapon_tec9,ct_grenade_hegrenade,t_grenade_hegrenade,ct_grenade_flashbang,t_grenade_flashbang,ct_grenade_smokegrenade,t_grenade_smokegrenade,ct_grenade_incendiarygrenade,t_grenade_incendiarygrenade,ct_grenade_molotovgrenade,t_grenade_molotovgrenade,ct_grenade_decoygrenade,t_grenade_decoygrenade,round_winner
15141,54.93,11.0,12.0,7,0,409.0,286.0,500.0,0.0,15250.0,9750.0,4.0,0.0,3.0,5.0,5.0,1.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,3.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,3.0,0.0,3.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0
66001,174.93,7.0,3.0,2,0,500.0,500.0,100.0,266.0,16850.0,26700.0,1.0,4.0,0.0,5.0,5.0,0.0,1.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1
115377,86.88,7.0,4.0,6,0,500.0,489.0,481.0,497.0,3350.0,5750.0,1.0,5.0,3.0,5.0,5.0,1.0,4.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,4.0,4.0,4.0,4.0,1.0,0.0,0.0,3.0,0.0,0.0,0
9409,34.34,10.0,5.0,2,1,190.0,120.0,100.0,100.0,450.0,1150.0,0.0,0.0,0.0,3.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1
44956,9.70,2.0,0.0,4,1,46.0,250.0,88.0,293.0,1150.0,2050.0,1.0,2.0,1.0,1.0,3.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,3.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,1
103151,175.00,8.0,7.0,3,0,500.0,500.0,400.0,0.0,700.0,4000.0,0.0,0.0,1.0,5.0,5.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,5.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,4.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0
6897,74.96,2.0,1.0,1,0,300.0,300.0,0.0,299.0,9850.0,10750.0,0.0,3.0,0.0,3.0,3.0,1.0,2.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,3.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,1.0,2.0,0.0,0.0,1
12180,74.95,5.0,12.0,1,0,344.0,472.0,100.0,477.0,9200.0,6600.0,0.0,5.0,0.0,4.0,5.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,3.0,0.0,0.0,0.0,0.0,...,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,2.0,0.0,0.0,1.0,0.0,0.0,0.0,2.0,0.0,0.0,0.0,0.0,0.0,1.0,0.0,1.0,1.0,0.0,0.0,0.0,1.0,0.0,2.0,0.0,2.0,0.0,0.0,0.0,3.0,0.0,0.0,1
371

In [68]:
#statistical summary 
df.describe()

,time_left,ct_score,t_score,map,bomb_planted,ct_health,t_health,ct_armor,t_armor,ct_money,t_money,ct_helmets,t_helmets,ct_defuse_kits,ct_players_alive,t_players_alive,ct_weapon_ak47,t_weapon_ak47,ct_weapon_aug,t_weapon_aug,ct_weapon_awp,t_weapon_awp,ct_weapon_bizon,t_weapon_bizon,ct_weapon_cz75auto,t_weapon_cz75auto,ct_weapon_elite,t_weapon_elite,ct_weapon_famas,t_weapon_famas,ct_weapon_g3sg1,t_weapon_g3sg1,ct_weapon_galilar,t_weapon_galilar,ct_weapon_glock,t_weapon_glock,ct_weapon_m249,t_weapon_m249,ct_weapon_m4a1s,t_weapon_m4a1s,...,t_weapon_p90,ct_weapon_r8revolver,t_weapon_r8revolver,ct_weapon_sawedoff,t_weapon_sawedoff,ct_weapon_scar20,t_weapon_scar20,ct_weapon_sg553,t_weapon_sg553,ct_weapon_ssg08,t_weapon_ssg08,ct_weapon_ump45,t_weapon_ump45,ct_weapon_xm1014,t_weapon_xm1014,ct_weapon_deagle,t_weapon_deagle,ct_weapon_fiveseven,t_weapon_fiveseven,ct_weapon_usps,t_weapon_usps,ct_weapon_p250,t_weapon_p250,ct_weapon_p2000,t_weapon_p2000,ct_weapon_tec9,t_weapon_tec9,ct_grenade_hegrenade,t_grenade_hegrenade,ct_grenade_flashbang,t_grenade_flashbang,ct_grenade_smokegrenade,t_grenade_smokegrenade,ct_grenade_incendiarygrenade,t_grenade_incendiarygrenade,ct_grenade_molotovgrenade,t_grenade_molotovgrenade,ct_grenade_decoygrenade,t_grenade_decoygrenade,round_winner
count,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.00000,117448.000000,117448.0,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.0,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.0,117448.000000,117448.000000,...,117448.000000,117448.0,117448.00000,117448.0,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.00000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000,117448.000000
mean,94.648097,6.769566,6.828775,3.520077,0.116511,408.522623,398.658828,317.751064,302.562300,10000.738625,11492.634613,2.120028,2.868512,1.650169,4.244363,4.235730,0.283547,1.286425,0.138725,0.007961,0.59285,0.332164,0.0,0.000085,0.156129,0.109955,0.003874,0.001831,0.114221,0.004904,0.0,0.000826,0.011299,0.114459,0.007195,3.260124,0.000009,0.0,0.065850,0.002835,...,0.000162,0.0,0.00006,0.0,0.000145,0.000749,0.000009,0.246569,0.853075,0.061534,0.021005,0.048243,0.03777,0.007842,0.000272,0.456747,0.394753,0.063407,0.009706,3.124140,0.126464,0.244832,0.289933,0.170816,0.005075,0.008344,0.029315,0.918866,0.447670,1.905320,1.895034,1.583084,1.671778,1.026318,0.020383,0.049605,1.386248,0.027689,0.025117,0.510362
std,53.224518,4.802249,4.832447,1.932589,0.320838,133.833268,141.393442,170.339769,174.118608,11308.757451,12245.826779,1.831718,1.976529,1.614360,1.221495,1.244582,0.591629,1.367108,0.466278,0.093354,0.67617,0.481009,0.0,0.009227,0.451898,0.386583,0.062121,0.042746,0.395475,0.072020,0.0,0.037924,0.113238,0.412851,0.088646,1.632670,0.002918,0.0,0.266809,0.054124,...,0.012718,0.0,0.00772,0.0,0.012030,0.027363,0.002918,0.560860,1.148492,0.269249,0.148075,0.234359,0.21491,0.089737,0.016504,0.845545,0.827029,0.292478,0.101792,1.588136,0.445489,0.540830,0.578560,0.421189,0.075922,0.096241,0.194610,1.287682,0.813887,1.769392,1.803067,1.738460,1.835046,1.462231,0.145991,0.231219,1.671632,0.169642,0.162253,0.499895
min,0.010000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0

It is good practice to scale the data after train test split why ?

if we don't split data before scaling then model will learn eventualy about test data

In [69]:
x = df.drop(columns='round_winner')
y = df['round_winner']

In [70]:
from sklearn.model_selection import train_test_split

In [71]:
x_train,x_test,y_train,y_test= train_test_split(x,y,test_size=0.20,random_state=42)

In [72]:
x_train.shape,x_test.shape,y_train.shape,y_test.shape

((93958, 96), (23490, 96), (93958,), (23490,))

In [73]:
from sklearn.preprocessing import StandardScaler

In [74]:
sc = StandardScaler()
x_train_scaled = sc.fit_transform(x_train)
x_test_scaled = sc.transform(x_test)

Make a model with our lda 

In [75]:
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import *


In [76]:
model = LogisticRegression()
model.fit(x_train_scaled,y_train)
y_pred = model.predict(x_test_scaled)
print(accuracy_score(y_test,y_pred))

0.759174116645381


### BUILD THE MODEL WITH LDA

In [77]:
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis

In [78]:
lda = LinearDiscriminantAnalysis()
lda.fit(x_train_scaled,y_train)


LinearDiscriminantAnalysis()

In [79]:
lda.transform(x_test_scaled)

array([[-0.69224519],
       [-0.72926778],
       [ 1.86900733],
       ...,
       [-0.412417  ],
       [ 0.62073932],
       [ 0.70458681]])

In [80]:
lda.coef_

array([[ 1.32450954e-01, -1.28879044e-02,  1.80798217e-02,
        -8.74488465e-02,  2.39884060e-01, -4.43098719e-01,
         3.27612026e-01, -6.05431498e-01,  6.33015335e-01,
        -2.14552694e-01,  1.68532336e-01,  2.93035188e-02,
         7.33365748e-02, -2.68716636e-02, -1.96820416e-01,
         3.65434345e-01, -1.93183010e-01,  6.07498388e-01,
        -2.03626114e-01,  3.76029060e-02, -3.08425645e-01,
         2.55454777e-01,  6.80258603e-17, -5.51711468e-04,
         1.05124538e-02, -6.89279741e-03, -2.86750520e-03,
         1.94754503e-02, -1.37777398e-01,  4.62222702e-02,
         1.53654309e-15,  1.84596657e-02, -4.97404086e-02,
         1.33798212e-01,  2.07171531e-02, -1.39679188e-01,
        -2.12532031e-16,  7.62876839e-16, -9.72800626e-02,
         4.80591752e-02, -4.53505704e-01,  1.04335981e-01,
        -9.27751790e-03,  1.14628755e-01, -1.59971534e-02,
         7.82742179e-03, -1.42257642e-02,  3.76839157e-02,
         6.27982602e-03, -6.33115926e-03, -1.20967801e-0

In [81]:
lda_coef = np.exp(np.abs(lda.coef_))

In [82]:
lda_coef

array([[1.14162302, 1.01297131, 1.01824425, 1.09138643, 1.27110177,
        1.55752608, 1.3876505 , 1.83204256, 1.88328075, 1.23930742,
        1.1835665 , 1.02973709, 1.07609266, 1.02723596, 1.21752537,
        1.44113982, 1.21310478, 1.83583311, 1.22583974, 1.03831884,
        1.36128029, 1.29104863, 1.        , 1.00055186, 1.0105679 ,
        1.00691661, 1.00287162, 1.01966633, 1.14772004, 1.04730717,
        1.        , 1.0186311 , 1.05099823, 1.14316212, 1.02093324,
        1.14990484, 1.        , 1.        , 1.10216901, 1.04923274,
        1.57381987, 1.10997332, 1.00932069, 1.12145703, 1.01612579,
        1.00785814, 1.01432743, 1.03840296, 1.00629959, 1.00635124,
        1.12858857, 1.03998677, 1.        , 1.00782618, 1.01787059,
        1.01507381, 1.02556116, 1.01876817, 1.        , 1.00634069,
        1.        , 1.01024978, 1.00138927, 1.01247098, 1.22364987,
        1.74599191, 1.08728405, 1.02362632, 1.05268431, 1.09877688,
        1.01636601, 1.00128427, 1.01376886, 1.00

In [83]:
lda_coef = lda_coef.flatten()

In [84]:
lda_coef

array([1.14162302, 1.01297131, 1.01824425, 1.09138643, 1.27110177,
       1.55752608, 1.3876505 , 1.83204256, 1.88328075, 1.23930742,
       1.1835665 , 1.02973709, 1.07609266, 1.02723596, 1.21752537,
       1.44113982, 1.21310478, 1.83583311, 1.22583974, 1.03831884,
       1.36128029, 1.29104863, 1.        , 1.00055186, 1.0105679 ,
       1.00691661, 1.00287162, 1.01966633, 1.14772004, 1.04730717,
       1.        , 1.0186311 , 1.05099823, 1.14316212, 1.02093324,
       1.14990484, 1.        , 1.        , 1.10216901, 1.04923274,
       1.57381987, 1.10997332, 1.00932069, 1.12145703, 1.01612579,
       1.00785814, 1.01432743, 1.03840296, 1.00629959, 1.00635124,
       1.12858857, 1.03998677, 1.        , 1.00782618, 1.01787059,
       1.01507381, 1.02556116, 1.01876817, 1.        , 1.00634069,
       1.        , 1.01024978, 1.00138927, 1.01247098, 1.22364987,
       1.74599191, 1.08728405, 1.02362632, 1.05268431, 1.09877688,
       1.01636601, 1.00128427, 1.01376886, 1.00102851, 1.03939

In [85]:
features_name = x.columns

In [86]:
df_features_imp = pd.DataFrame({'Features':features_name,'Imp_val':lda_coef})

In [87]:
df_features_imp

,Features,Imp_val
0,time_left,1.141623
1,ct_score,1.012971
2,t_score,1.018244
3,map,1.091386
4,bomb_planted,1.271102
...,...,...
91,t_grenade_incendiarygrenade,1.009694
92,ct_grenade_molotovgrenade,1.010343
93,t_grenade_molotovgrenade,1.104343
94,ct_grenade_decoygrenade,1.000688


In [88]:
top_20_features = df_features_imp.nlargest(20,'Imp_val')

In [89]:
top_20_features

,Features,Imp_val
8,t_armor,1.883281
17,t_weapon_ak47,1.835833
7,ct_armor,1.832043
65,t_weapon_sg553,1.745992
40,ct_weapon_m4a4,1.573820
5,ct_health,1.557526
15,t_players_alive,1.441140
6,t_health,1.387650
20,ct_weapon_awp,1.361280
21,t_weapon_awp,1.291049


In [90]:
imp_col_index = top_20_features.index

In [91]:
imp_col_index

Index([8, 17, 7, 65, 40, 5, 15, 6, 20, 21, 4, 89, 9, 18, 64, 14, 16, 87, 10,
       35],
      dtype='int64')

In [92]:
x_train_scaled = x_train_scaled[:,imp_col_index]
x_test_scaled = x_test_scaled[:,imp_col_index]

### Logistic Regression

In [95]:
log_model = LogisticRegression()
log_model.fit(x_train_scaled,y_train)
y1 = log_model.predict(x_test_scaled)
accuracy_score(y_test,y1)

0.7589186888037462

### Decision Tree

In [97]:
from sklearn.tree import DecisionTreeClassifier
dt = DecisionTreeClassifier()
dt.fit(x_train_scaled,y_train)
y2 = dt.predict(x_test_scaled)
accuracy_score(y_test,y2)

0.8094082588335462

### Random Forest


In [98]:
from sklearn.ensemble import RandomForestClassifier
rf = RandomForestClassifier()
rf.fit(x_train_scaled,y_train)
y3 = rf.predict(x_test_scaled)
accuracy_score(y_test,y3)

0.8578118348233291